modules/account-lib/test/unit/coin/dot/transaction.ts

```diff
@@ -0,0 +1,54 @@
+import should from 'should';
+import { coins } from '@bitgo/statics';
+import * as DotResources from '../../../resources/dot';
+import { KeyPair, Transaction, TransferBuilder } from '../../../../src/coin/dot';
+
+describe('Dot Transaction', () => {
+  let tx: Transaction;
+
+  beforeEach(() => {
+    const config = coins.get('dot');
+    tx = new Transaction(config);
+  });
+
+  describe('empty transaction', () => {
+    it('should throw empty transaction', () => {
+      should.throws(() => tx.toJson(), 'Empty transaction');
+      should.throws(() => tx.toBroadcastFormat(), 'Empty transaction');
+    });
+
+    it('should not sign', async () => {
+      try {
+        await tx.sign(new KeyPair({ prv: DotResources.accounts.account1.secretKey }));
+      } catch (e) {
+        should.equal(e.message, 'No transaction data to sign');
+      }
+    });
+  });
+
+  describe('sign transaction', () => {
+    it('cannot sign - wrong account secret', () => {
+      tx.sender(DotResources.accounts.account1.address);
+      should.deepEqual(tx.canSign({ key: DotResources.accounts.account2.secretKey }), false);
+    });
+
+    it('can sign', () => {
+      tx.sender(DotResources.accounts.account2.address);
+      should.deepEqual(tx.canSign({ key: DotResources.accounts.account2.secretKey }), true);
+    });
+  });
+
+  describe('should build from raw unsigned tx', async () => {
+    it('Transaction size validation', async () => {
```

Comment:
We Should add failure cases for invalid size?

Reply:
Builds a tx to verify, i dont think we can build a tx with an invalid size